src/model_io/urdf/src/URDFModelExport.cpp

```diff
@@ -241,6 +242,68 @@ bool exportJoint(IJointConstPtr joint, LinkConstPtr parentLink, LinkConstPtr chi
     return ok;
 }
 
+/*
+ * Add a <link> and <joint> URDF elements to the specified parent element for the specified additional frame.
+ *
+ * URDF does not have the concept of frame (see https://discourse.ros.org/t/urdf-ng-link-and-frame-concepts/56),
+ * so for each additional frame we need to add a URDF "fake" mass-less link.
+ *
+ * If parent_element contains a pointer to a tag <parent_element></parent_element>, this function modifies it to be something like:
+ *
+ * <parent_element>
+ *   <link name="<frame_name>"/>
+ *   <joint name="<frame_name>_fixed_joint" type="fixed">
+ *      <origin xyz="0.1 0.2 0.3" rpy="0 0 3.1416"/>
+ *      <parent link="<link_name>"/>
+ *      <child link="<frame_name>"/>
+ *   </joint>
+ * <parent_element/>
+ *
+ * where the actual values of the added element and attributes depend on the input frame_name, link_H_frame, parent_link_name and direction_options arguments.
+ */
+enum exportAdditionalFrameDirectionOption {
+    FAKE_LINK_IS_CHILD,
+    FAKE_LINK_IS_PARENT
+};
+bool exportAdditionalFrame(const std::string frame_name, Transform link_H_frame, const std::string link_name, exportAdditionalFrameDirectionOption direction_option, xmlNodePtr parent_element)
+{
+    bool ok=true;
+
+    // Export fake link
+    xmlNodePtr link_xml = xmlNewChild(parent_element, NULL, BAD_CAST "link", NULL);
+    xmlNewProp(link_xml, BAD_CAST "name", BAD_CAST frame_name.c_str());
```

Comment:
If any future developer asks why there is `BAD_CAST` all across the code, we could possibly document this https://stackoverflow.com/questions/45058878/why-libxml2-uses-bad-cast-everywhere-in-a-c-c-code somewhere in the code ?

Reply:
Fixed in https://github.com/robotology/idyntree/pull/557/commits/ee923f4c3d078ff363f9cbe36fc1f31e755ff35b .